source/connection-string/connection-string-spec.md

```diff
@@ -216,13 +216,25 @@ The values in connection options MUST be URL decoded by the parser. The values c
   ```
 
 - Key value pairs: A value that represents one or more key and value pairs. Multiple key value pairs are delimited by a
-  comma (","). The key is everything up to the first colon sign (":") and the value is everything afterwards. If any
-  keys or values containing a comma (",") or a colon (":") they must be URL encoded. For example:
+  comma (","). The key is everything up to the first colon sign (":") and the value is everything afterwards.
+
+  For example:
 
   ```
   ?readPreferenceTags=dc:ny,rack:1
   ```
 
+  Drivers MUST handle unencoded colon signs (":") within the value. For example, given the encoded input:
+
+  ```
+  ?authMechanismProperties=TOKEN_RESOURCE%3Amongodb%3A%2F%2Ffoo
+  ```
+
+  the driver MUST interpret the key as `TOKEN_RESOURCE` and the value as `mongodb://foo`.
+
+  Drivers MUST document that if keys or values contain a comma (",") they MUST not be provided as part of the connection
+  string, since it would interfere with parsing.
```

Comment:
```suggestion
  For any option key-value pair that may contain a comma (such as `TOKEN_RESOURCE`), drivers MUST document that: a value containing a comma (",") MUST NOT be provided as part of the connection
  string. This prevents use of values that would interfere with parsing.
```
1. Clarifies that we do not want the "since" clause to be part of the user-facing docs.
2. Are users allowed to specify arbitrary keys? I thought we controlled them all.
3. A general comment seems like it would be confusing. We use the term "value" to refer (at least) to both: option values, and option key-value pair values. Option values can have commas, but I'm not sure how to express the distinction in user docs without using a convoluted phrase like "option key-value pair values".

It also seems fine (preferable?) to not add this new paragraph/requirement to the connection string spec at this time, since it is already quite nicely phrased in the auth spec for `TOKEN_RESOURCE` in particular.